src/AbstractProxyFixer.php

```diff
@@ -72,19 +97,26 @@ public function supports(\SplFileInfo $file)
     public function setWhitespacesConfig(WhitespacesFixerConfig $config)
     {
         parent::setWhitespacesConfig($config);
-        $this->proxyFixer->setWhitespacesConfig($config);
+
+        foreach ($this->proxyFixers as $fixer) {
+            if ($fixer instanceof WhitespacesAwareFixerInterface) {
+                $fixer->setWhitespacesConfig($config);
+            }
+        }
     }
 
     /**
      * {@inheritdoc}
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $this->proxyFixer->fix($file, $tokens);
+        foreach ($this->proxyFixers as $fixer) {
+            $fixer->fix($file, $tokens);
```

Comment:
shouldn't the `supports(\SplFileInfo $file)` be done here for each fixer, as the previous check has an early return with `true`?

Reply:
I'm not sure I understand your comment. Each proxied fixer is going to do this check internally when calling `fix()`, isn't?

Reply:
I dropped on it as well @SpacePossum (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3149#discussion_r144108084)

`fix` is calling `tokens->count`, `supports` and `isCandidate`

Reply:
ah right, so the abstract fixer could you return `true` for `supports` and `isCandidate` without asking the proxied fixers?

Reply:
in theory, yes. it would work fine.
yet, someone external could depends on `supports`/`isCandidate` as well, so we need to provide good data. this case was discussed and it's problematic to have them to be called by outsiders before calling `->fix`, so we call them in `->fix` as well, and they will be dropped in v3 hopefully